mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/BlockMatrix.scala

```diff
@@ -456,15 +461,20 @@ class BlockMatrix @Since("1.3.0") (
    * with each other.
    */
   @Since("1.3.0")
-  def multiply(other: BlockMatrix): BlockMatrix = {
+  def multiply(
```

Comment:
you shouldn't change the signature of a public method and still call it `@Since(1.3.0)`. Maybe move the inside of this to 
```scala
private def multiplyImpl(
  other: BlockMatrix,
  shufflePartitioner: GridPartitioner,
  midDimPartNum: Int,
  resultPartitioner: GridPartitioner): BlockMatrix
```

but keep the
```scala
@Since("1.3.0")
def multiply(other: BlockMatrix): BlockMatrix = {
  multiplyImpl(other, ...)
}
```